src/isomorphic/classic/types/ReactPropTypes.js

```diff
@@ -144,6 +144,11 @@ function createAnyTypeChecker() {
 
 function createArrayOfTypeChecker(typeChecker) {
   function validate(props, propName, componentName, location, propFullName) {
+    if (typeof typeChecker !== 'function') {
+      return new Error(
+        `Invalid argument \`${propFullName}\` supplied to \`${componentName}\`, expected valid PropType.`
```

Comment:
This message isn't awesome. You're actually trying to say that the component author wrote a bad propType spec, but the way this error will show sounds like the user provided an invalid prop value. Can we tune it so that's more clear?